src/include/migraphx/op/select_module.hpp

```diff
@@ -43,25 +43,49 @@ struct select_module
 
     std::string name() const { return "select_module"; }
 
-    shape compute_shape(const std::vector<shape>&, const std::vector<module_ref>&) const
+    shape compute_shape(const std::vector<shape>& inputs, const std::vector<module_ref>&) const
     {
+        check_shapes{inputs, *this, true}.has_at_least(1);
         return shape{output_dyn_shapes};
     }
 
+    std::vector<std::string> get_input_parameter_names(module_ref mod) const
+    {
+        auto param_names = mod->get_parameter_names();
+        std::vector<std::string> ret;
+        std::copy_if(param_names.cbegin(),
+                     param_names.cend(),
+                     std::back_inserter(ret),
+                     [](auto pn) { return not contains(pn, "#output_"); });
```

Comment:
```suggestion
                     [](auto pn) { return not starts_with(pn, "#output_"); });
```

Reply:
Output parameters start with the module name, so this wouldn't work.